### PR TITLE
fix: slug value in webhook payload

### LINKED ISSUE #12285: Webhook events send the event type title rather than the documented event type slug in the "type" field
### Issue Summary

The docs here https://cal.com/docs/core-features/webhooks#adding-a-custom-payload-template state that the `type` field is `The event type slug`, yet I am getting back the event type title.

### Steps to Reproduce

1. Create new webhook
2. Point to testing site (https://webhook.site)
3. Create new meeting
4. Look at webhook event and notice the `type` field is the event type title and not the event type slug.

### Actual Results

Webhooks return event type title in the type field.

### Expected Results

Webhooks return event type slug in the type field.

### Technical details

Tested on cal.com hosted version.

### Evidence

Create a new account on cal.com and then follow the steps to reproduce above.

The event type:
Screenshot 2023-11-08 at 3 35 52 PM

The event:
Screenshot 2023-11-08 at 3 31 49 PM

### PR DESCRIPTION
**Pull request by manpoffc**
## What does this PR do?
This PR fixes the slug value in the payload of webhook which was earlier returning the value of event title.

Fixes #12285  

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

File: packages/features/bookings/lib/handleCancelBooking.ts [hunk 1]
```diff
@@ -224,7 +224,7 @@ async function handler(req: CustomRequest) {
 
   const evt: CalendarEvent = {
     title: bookingToDelete?.title,
-    type: (bookingToDelete?.eventType?.title as string) || bookingToDelete?.title,
+    type: (bookingToDelete?.eventType?.slug as string) || bookingToDelete?.title,
```
Comment (nicolls1) [inline packages/features/bookings/lib/handleCancelBooking.ts:227]: Should the second part also be changed to slug like:
```
type: (bookingToDelete?.eventType?.slug as string) || bookingToDelete?.slug,
```

**Review by nicolls1 — COMMENTED** [review]
Thank you for putting this up! Had one question

File: packages/features/bookings/lib/handleCancelBooking.ts (hunk 1 above)
Comment (manpoffc) [reply packages/features/bookings/lib/handleCancelBooking.ts:227]: Certainly, I missed onto that one.

**Review by CarinaWolli — DISMISSED** [review]
We also need to fix e2e and unit tests for that, could you take a look at that? 🙏

**Comment by manpoffc** [comment]
I am actually facing issue in running the test cases, can anyone assist me with the e2e tests so that i can resolve this issue asap?

**Comment by manpoffc** [comment]
I've updated the Booking Triggers webhook values. Unable to run tests locally, so uncertain about the failing test case. Can you guide me on resolving this? Thanks :)

File: apps/web/test/utils/bookingScenario/bookingScenario.ts [hunk 2]
```diff
@@ -1261,7 +1261,7 @@ export function getExpectedCalEventForBookingRequest({
 }) {
   return {
     // keep adding more fields as needed, so that they can be verified in all scenarios
-    type: eventType.title,
+    type: eventType.slug,
```
Comment (CarinaWolli) [inline apps/web/test/utils/bookingScenario/bookingScenario.ts:1264]: @manpoffc this fixed the unit test

**Review by CarinaWolli — APPROVED** [review]
Changes look good and tests are passing now 🙏

File: apps/web/test/utils/bookingScenario/bookingScenario.ts (hunk 2 above)
Comment (manpoffc) [reply apps/web/test/utils/bookingScenario/bookingScenario.ts:1264]: Thank you for the help :)

**Comment by Shpadoinkle** [comment]
was there an announcement for this change? 

I'd like to add that although the documentation did not match the actual responses from webhooks. 
i would treat a change like this more carefully almost like a breaking change and give proper warning to developers,  as our current webhook integrations failed after this latest version updates as they were built off of the actual responses we were getting from the webhooks verse the documentation.

**Comment by manpoffc** [comment]
> was there an announcement for this change?
> 
> I'd like to add that although the documentation did not match the actual responses from webhooks. i would label a change like this more carefully almost like a breaking change not just a fix. and give proper warning to developers, as our current webhook integrations failed after this latest version updates as they were built off of the actual responses we were getting from the webhooks verse the documentation.

Thank you so much for this, I missed onto the documentation part and raise a new issue and change the documentation right away.

**Comment by Shpadoinkle** [comment]
The current documentation is fine from what i can see.   

It's just that the change from webhook integrations currently returning a title prior to v3.5.1, to now getting a different response could be breaking if they weren't checking.

**Pull request merged**